src/com/example/Demo_Subj/RoomView.java

```diff
@@ -41,34 +162,207 @@ protected void onDraw(Canvas canvas){
 
         // find out which room to draw
         //TODO should be replaced by a proper find() in the list
-        for (Iterator<Room> iter = roomList.iterator(); iter.hasNext(); ) {
-            Room room = iter.next();
-            if (room.getRoomID() == GlobalInformation.getCurrentRoom()) {
-                drawRoom = room;
+
+        if(lastRoom != aktRoomID) {                                                 //hat sich lastroom gegenüber dem letzten Aufruf geändert? wenn ja dann lade den aktuellen Raum
+            for (Iterator<Room> iter = World.getAllRooms().iterator(); iter.hasNext(); ) {
```

Comment:
Hier muss die (von Rene später befüllte) TreeMap benutzt werden, die List<Room> in der World muss weg!